### PR TITLE
[REVIEW] Handle `axis` string keywords in `concat`

### LINKED ISSUE #2487: [BUG] cudf.concat acts wrong when axis='columns'
**Describe the bug**

For pandas, concat axis='columns' is identical to axis=1. But cudf acts different when axis='columns'.

**Steps/Code to reproduce bug**

```
In [1]: import numpy as np                                                                                                                                                                                         

In [2]: import pandas as pd                                                                                                                                                                                        

In [3]: import cudf                                                                                                                                                                                                

In [4]: a = pd.DataFrame(np.random.randint(10, size=(5, 3)), columns=[1, 2, 3])                                                                                                                                    

In [5]: a                                                                                                                                                                                                          
Out[5]: 
   1  2  3
0  9  9  1
1  6  6  3
2  2  5  5
3  1  1  4
4  5  7  7

In [6]: b = pd.DataFrame(np.random.randint(10, size=(5, 4)), columns=[4, 5, 6, 7])                                                                                                                                 

In [7]: b                                                                                                                                                                                                          
Out[7]: 
   4  5  6  7
0  1  3  1  0
1  8  3  8  9
2  2  0  7  2
3  7  8  2  7
4  2  2  8  2

In [8]: pd.concat([a, b], axis='columns')                                                                                                                                                                          
Out[8]: 
   1  2  3  4  5  6  7
0  9  9  1  1  3  1  0
1  6  6  3  8  3  8  9
2  2  5  5  2  0  7  2
3  1  1  4  7  8  2  7
4  5  7  7  2  2  8  2

In [9]: ca = cudf.DataFrame.from_pandas(a)                                                                                                                                                                         

In [10]: cb = cudf.DataFrame.from_pandas(b)                                                                                                                                                                        

In [11]: cudf.concat([ca, cb], axis='columns')  # wrong                                                                                                                                                            
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-11-b6c1cf89e541> in <module>
----> 1 cudf.concat([ca, cb], axis='columns')  # wrong

~/miniconda3/lib/python3.7/site-packages/cudf/multi.py in concat(objs, axis, ignore_index)
     50 
     51     if typ is DataFrame:
---> 52         return DataFrame._concat(objs, axis=axis, ignore_index=ignore_index)
     53     elif typ is Series:
     54         return Series._concat(objs, axis=axis)

~/miniconda3/lib/python3.7/site-packages/cudf/dataframe/dataframe.py in _concat(cls, objs, axis, ignore_index)
   1268         if len(set(frozenset(o.columns) for o in objs)) != 1:
   1269             what = set(frozenset(o.columns) for o in objs)
-> 1270             raise ValueError('columns mismatch: {}'.format(what))
   1271 
   1272         objs = [o for o in objs]

ValueError: columns mismatch: {frozenset({1, 2, 3}), frozenset({4, 5, 6, 7})}

In [12]: cudf.concat([ca, cb], axis=1)  # right                                                                                                                                                                    
Out[12]: <cudf.DataFrame ncols=7 nrows=5 >

In [13]: _.to_pandas()                                                                                                                                                                                             
Out[13]: 
   1  2  3  4  5  6  7
0  9  9  1  1  3  1  0
1  6  6  3  8  3  8  9
2  2  5  5  2  0  7  2
3  1  1  4  7  8  2  7
4  5  7  7  2  2  8  2
```

**Expected behavior**

When axis='columns', it acts like axis=1.

**Environment overview (please complete the following information)**
 - Environment location: Bare-metal
 - Method of cuDF install: conda

**Environment details**
Please run and paste the output of the `cudf/print_env.sh` script here, to gather any other relevant environment details

**git***
Not inside a git repository

***OS Information***
Alibaba Group Enterprise Linux Server release 7.2 (Paladin)
NAME="Alibaba Group Enterprise Linux Server"
VERSION="7.2 (Paladin)"
ID="alios"
ID_LIKE="fedora"
VERSION_ID="7.2"
PRETTY_NAME="Alibaba Group Enterprise Linux Server 7.2 (Paladin)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:alibaba:enterprise_linux:7.2:GA:server"
HOME_URL="https://www.aliyun.com/"
BUG_REPORT_URL="http://bbs.aliyun.com/"

ALIBABA_BUGZILLA_PRODUCT="Alibaba Group Enterprise Linux 7"
ALIBABA_BUGZILLA_PRODUCT_VERSION=7.2
ALIBABA_SUPPORT_PRODUCT="Alibaba Group Enterprise Linux"
ALIBABA_SUPPORT_PRODUCT_VERSION=7.2
Alibaba Group Enterprise Linux Server release 7.2 (Paladin)
Alibaba Group Enterprise Linux Server release 7.2 (Paladin)
Linux tb1a01277.sqa.tbc 3.10.0-327.ali2012.alios7.x86_64 #1 SMP Mon Oct 9 14:09:14 CST 2017 x86_64 x86_64 x86_64 GNU/Linux

***GPU Information***
Wed Aug  7 23:24:05 2019       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 418.67       Driver Version: 418.67       CUDA Version: 10.1     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  Tesla V100-SXM2...  Off  | 00000000:1B:00.0 Off |                    0 |
| N/A   31C    P0    39W / 300W |      0MiB / 16130MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   1  Tesla V100-SXM2...  Off  | 00000000:1C:00.0 Off |                    0 |
| N/A   32C    P0    41W / 300W |      0MiB / 16130MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   2  Tesla V100-SXM2...  Off  | 00000000:3D:00.0 Off |                    0 |
| N/A   34C    P0    40W / 300W |      0MiB / 16130MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   3  Tesla V100-SXM2...  Off  | 00000000:3E:00.0 Off |                    0 |
| N/A   33C    P0    52W / 300W |  15554MiB / 16130MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   4  Tesla V100-SXM2...  Off  | 00000000:B1:00.0 Off |                    0 |
| N/A   32C    P0    38W / 300W |      0MiB / 16130MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   5  Tesla V100-SXM2...  Off  | 00000000:B2:00.0 Off |                    0 |
| N/A   33C    P0    38W / 300W |      0MiB / 16130MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   6  Tesla V100-SXM2...  Off  | 00000000:DB:00.0 Off |                    0 |
| N/A   31C    P0    38W / 300W |      0MiB / 16130MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   7  Tesla V100-SXM2...  Off  | 00000000:DC:00.0 Off |                    0 |
| N/A   31C    P0    39W / 300W |      0MiB / 16130MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|    3     20821      C   ...naconda3/envs/tensorflow_lcl/bin/python 15543MiB |
+-----------------------------------------------------------------------------+

***CPU***
Architecture:          x86_64
CPU op-mode(s):        32-bit, 64-bit
Byte Order:            Little Endian
CPU(s):                96
On-line CPU(s) list:   0-95
Thread(s) per core:    2
Core(s) per socket:    24
Socket(s):             2
NUMA node(s):          1
Vendor ID:             GenuineIntel
CPU family:            6
Model:                 85
Model name:            Intel(R) Xeon(R) Platinum 8163 CPU @ 2.50GHz
Stepping:              4
CPU MHz:               2499.902
CPU max MHz:           3100.0000
CPU min MHz:           1000.0000
BogoMIPS:              4998.90
Virtualization:        VT-x
Hypervisor vendor:     vertical
Virtualization type:   full
L1d cache:             32K
L1i cache:             32K
L2 cache:              1024K
L3 cache:              33792K
NUMA node0 CPU(s):     0-95
Flags:                 fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc aperfmperf eagerfpu pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch ida arat epb pln pts dtherm tpr_shadow vnmi flexpriority ept vpid fsgsbase tsc_adjust bmi1 hle avx2 smep bmi2 erms invpcid rtm cqm mpx avx512f avx512dq rdseed adx smap clflushopt avx512cd avx512bw avx512vl xsaveopt xsavec xgetbv1 cqm_llc cqm_occup_llc

***CMake***
which: no cmake in (/sbin:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/usr/X11R6/bin:/home/tops/bin:/home/xuye.qin/.local/bin:/home/xuye.qin/bin)

***g++***
/usr/bin/g++
g++ (GCC) 4.8.5 20150623 (Red Hat 4.8.5-28)
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.


***nvcc***
which: no nvcc in (/sbin:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/usr/X11R6/bin:/home/tops/bin:/home/xuye.qin/.local/bin:/home/xuye.qin/bin)

***Python***
/usr/bin/python
Python 2.7.5

***Environment Variables***
PATH                            : /sbin:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/usr/X11R6/bin:/home/tops/bin:/home/xuye.qin/.local/bin:/home/xuye.qin/bin
LD_LIBRARY_PATH                 : :/usr/local/cuda/lib64
NUMBAPRO_NVVM                   : 
NUMBAPRO_LIBDEVICE              : 
CONDA_PREFIX                    : 
PYTHON_PATH                     : 

conda not found
pip not found

### PR DESCRIPTION
**Pull request by kkraus14**
Fixes #2487

### CODE REVIEW

File: python/cudf/cudf/core/reshape.py [hunk 1]
```diff
@@ -35,6 +37,16 @@ def concat(objs, axis=0, ignore_index=False, sort=None):
...
     allowed_typs = {Series, DataFrame}
+
+    if isinstance(axis, str):
+        axis = _axis_map.get(axis, None)
+        if axis is None:
+            raise ValueError(
+                '`axis` must be 0 / "index" or 1 / "columns", got: {0}'.format(
+                    axis
+                )
+            )
```
Comment (kkraus14) [inline python/cudf/cudf/core/reshape.py]: Need to raise this before overwriting axis

Comment (kkraus14) [reply python/cudf/cudf/core/reshape.py]: Also need to handle when integer input isn't 0 / 1

File: python/cudf/cudf/tests/test_multi.py [hunk 2]
```diff
@@ -46,7 +46,8 @@ def make_frames(index=None, nulls="none"):
```
Comment (thomcom) [inline python/cudf/cudf/tests/test_multi.py]: Suggest renaming the `test_multi.py` file to `test_concat.py`. `multi.py` is no longer in this repo.

Comment (kkraus14) [reply python/cudf/cudf/tests/test_multi.py]: Done

**Comment by kkraus14** [comment]
rerun tests

**Pull request merged**